modules/src/main/modules/elasticsearch/module.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Hibernate Search, full-text search for your domain model
+ ~
+ ~ License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ ~ See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+  -->
+<module xmlns="urn:jboss:module:1.1" name="org.hibernate.search.elasticsearch-client" slot="${hibernate.search.version}">
+    <resources>
+        <resource-root path="rest-${elasticsearchClientVersion}.jar"/>
+        <resource-root path="sniffer-${elasticsearchClientVersion}.jar"/>
+        <resource-root path="gson-${elasticsearchGsonVersion}.jar"/>
+    </resources>
+    <dependencies>
+        <module name="org.apache.httpcomponents"/>
+        <module name="org.apache.commons.logging"/>
```

Comment:
But I think we still need to add a dependency to a Jackson module here (even if we leave it in `hibernate-search-elasticsearch`). The class `org.elasticsearch.client.sniff.ElasticsearchHostsSniffer` depends on Jackson classes to parse JSON.

... unless modules visible by the `hibernate-search-elasticsearch` module are automatically visible by its dependencies?

Now which module to depend on, I'm not sure. I think `org.jboss.resteasy.resteasy-jackson-provider` would do the trick, but maybe you'll want to be more precise and just depend on `org.codehaus.jackson.jackson-core-asl`, which should be enough.